### PR TITLE
Stripe success flow

### PR DESCRIPTION
**Pull request by dem4ron**
- Add redirect link, remove success modal
- Copy paste welcome modal
- Tweaks

### CODE REVIEW

File: app/helpers/react_components/modals/welcome_insiders_modal.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module ReactComponents
+  module Modals
+    class WelcomeInsidersModal < ReactComponent
```
Comment (ErikSchierboom) [inline app/helpers/react_components/modals/welcome_insiders_modal.rb]: Maybe rename to:

```suggestion
    class WelcomeToInsidersModal < ReactComponent
```

File: app/helpers/react_components/modals/welcome_insiders_modal.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+          {
+            endpoint: Exercism::Routes.hide_api_settings_introducer_path(slug)
+          }
+        )
+      end
+
+      private
+      def slug
+        "welcome-insiders-modal"
+      end
```
Comment (ErikSchierboom) [inline app/helpers/react_components/modals/welcome_insiders_modal.rb]:
```suggestion
      def slug = "welcome-insiders-modal"
```

File: app/helpers/react_components/modals/welcome_premium_modal.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+          {
+            endpoint: Exercism::Routes.hide_api_settings_introducer_path(slug)
+          }
+        )
+      end
+
+      private
+      def slug
+        "welcome-premium-modal"
+      end
```
Comment (ErikSchierboom) [inline app/helpers/react_components/modals/welcome_premium_modal.rb]:
```suggestion
      def slug = "welcome-premium-modal"
```

File: app/helpers/react_components/modals/welcome_premium_modal.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+module ReactComponents
+  module Modals
+    class WelcomePremiumModal < ReactComponent
```
Comment (ErikSchierboom) [inline app/helpers/react_components/modals/welcome_premium_modal.rb]: Maybe rename to:

```suggestion
    class WelcomeToPremiumModal < ReactComponent
```

File: app/helpers/react_components/modals/welcome_insiders_modal.rb (hunk 1 above)
Comment (dem4ron) [reply app/helpers/react_components/modals/welcome_insiders_modal.rb]: I feel it's a bit redundant. I'd prefer the classnames to match the slug, React component, and component key. what do you think @iHiD?

File: app/helpers/react_components/premium/price_option.rb [hunk 5]
```diff
@@ -8,7 +8,8 @@ def to_s
           "premium-price-option", {
             user_signed_in: user_signed_in?,
             captcha_required: !current_user || current_user.captcha_required?,
-            recaptcha_site_key: ENV.fetch('RECAPTCHA_SITE_KEY', Exercism.secrets.recaptcha_site_key)
+            recaptcha_site_key: ENV.fetch('RECAPTCHA_SITE_KEY', Exercism.secrets.recaptcha_site_key),
+            premium_redirect_link: Exercism::Routes.premium_index_path
```
Comment (ErikSchierboom) [inline app/helpers/react_components/premium/price_option.rb]: If I test this locally, this route does not exist. `premium_path` does

File: app/helpers/react_components/modals/welcome_insiders_modal.rb (hunk 1 above)
Comment (ErikSchierboom) [reply app/helpers/react_components/modals/welcome_insiders_modal.rb]: > I feel it's a bit redundant

Well, we also have a `CopyToClipboardButton` component and not `CopyClipboardButton` 🤷 I won't die on this hill, but I like explicitness.

> I'd prefer the classnames to match the slug, React component, and component key.

Well, these could also be renamed? But feel free to ignore for now, it's low priority.

File: app/helpers/view_components/site_header.rb [hunk 6]
```diff
@@ -62,7 +62,7 @@ def signed_in_nav
               generic_nav("Contribute", submenu: CONTRIBUTE_SUBMENU, path: Exercism::Routes.contributing_root_path, offset: 20),
               generic_nav("Community", submenu: COMMUNITY_SUBMENU, path: Exercism::Routes.community_path, offset: 0),
               # generic_nav("Resources", submenu: LEARN_SUBMENU, offset: 100),
-              generic_nav("Premium", path: Exercism::Routes.donate_path, offset: 150, view: :mentoring),
+              generic_nav("Premium", path: Exercism::Routes.premium_index_path, offset: 150, view: :mentoring),
```
Comment (ErikSchierboom) [inline app/helpers/view_components/site_header.rb]: What does `view: :mentoring` do?

File: app/helpers/react_components/premium/price_option.rb (hunk 5 above)
Comment (dem4ron) [reply app/helpers/react_components/premium/price_option.rb]: yes, it is an update by Jeremy

**Pull request merged**